internal/types/path_test.go

```diff
@@ -220,10 +178,63 @@ func TestRemoveByPath(t *testing.T) {
 			t.Parallel()
 
 			doc := sourceDoc.DeepCopy()
-			doc.RemoveByPath(tc.path...)
+			RemoveByPath(doc, tc.path...)
 			assert.Equal(t, tc.res, doc, tc.name)
 		})
 	}
+
+	t.Run("test array remove by path", func(t *testing.T) {
+		t.Parallel()
+
+		src := must.NotFail(NewArray(
+			"0", float64(42.13), int32(1000), "2",
+			must.NotFail(NewDocument(
+				"document", "abc",
+				"score", float64(42.13),
+				"age", int32(1000),
+				"foo", deepDoc.DeepCopy(),
+			)),
+			must.NotFail(NewArray("1", "2", "3")),
+		))
+
+		type testCase struct {
+			name     string
+			path     []string
+			expected *Array
+		}
+
+		for _, tc := range []testCase{
+			{
+				name:     "array: remove by path",
+				path:     []string{"4"},
+				expected: must.NotFail(NewArray("0", float64(42.13), int32(1000), "2", must.NotFail(NewArray("1", "2", "3")))),
+			},
+			{
+				name:     "array: index exceded",
+				path:     []string{"11"},
+				expected: src.DeepCopy(),
+			},
+			{
+				name:     "array: empty path",
+				path:     []string{},
+				expected: src.DeepCopy(),
+			},
+			{
+				name:     "array: index is not number",
+				path:     []string{"abcd"},
+				expected: src.DeepCopy(),
+			},
+		} {
+			tc := tc
+			t.Run(fmt.Sprint(tc.path), func(t *testing.T) {
```

Comment:
We can use here `name` instead of `fmt.Sprintf(tc.path)` as test name

Reply:
nice catch! done for both cases.